libraries/src/Application/CMSApplication.php

```diff
@@ -904,40 +904,72 @@ public function login($credentials, $options = [])
             $authorisations = $authenticate->authorise($response, $options);
             $denied_states  = Authentication::STATUS_EXPIRED | Authentication::STATUS_DENIED;
 
-            foreach ($authorisations as $authorisation) {
-                if ((int) $authorisation->status & $denied_states) {
-                    // Trigger onUserAuthorisationFailure Event.
-                    $dispatcher->dispatch('onUserAuthorisationFailure', new AuthorisationFailureEvent('onUserAuthorisationFailure', [
-                        'subject' => (array) $authorisation,
-                        'options' => $options,
-                    ]));
-
-                    // If silent is set, just return false.
-                    if (isset($options['silent']) && $options['silent']) {
-                        return false;
-                    }
+            // Validate that $authorisations is an array before iterating
+            if (!is_array($authorisations)) {
+                // Log the issue for debugging purposes
+                \JLog::add(
+                    sprintf(
+                        'Invalid data type for $authorisations in %s on line %d. Expected array, got %s.',
+                        __FILE__,
+                        __LINE__,
+                        gettype($authorisations)
+                    ),
+                    \JLog::WARNING,
+                    'api'
+                );
 
-                    // Return the error.
-                    switch ($authorisation->status) {
-                        case Authentication::STATUS_EXPIRED:
-                            $this->enqueueMessage(Text::_('JLIB_LOGIN_EXPIRED'), 'error');
+                // Set $authorisations to an empty array to avoid further issues
+                $authorisations = [];
+            }
 
+            foreach ($authorisations as $authorisation) {
+                // Ensure $authorisation is an object and has the 'status' property
+                if (is_object($authorisation) && property_exists($authorisation, 'status')) {
```

Comment:
```suggestion
                if (\is_object($authorisation) && property_exists($authorisation, 'status')) {
```
PHPCS global namespace (2)